qiskit/transpiler/basepasses.py

```diff
@@ -131,7 +131,10 @@ def __call__(self, circuit, property_set=None):
             result_circuit = circuit.copy()
 
         if self.property_set["layout"]:
-            result_circuit._layout = self.property_set["layout"]
+            result_circuit._layout = (
+                self.property_set["layout"],
+                self.property_set["original_qubit_indices"],
+            )
```

Comment:
Even with this being internal, I think it might be a good idea to make this an attribute-based lookup object rather than a tuple-like one, so if there are any issues caused by changing the type, we only have to suffer the pain once.

For example, I could easily imagine that we might want to expand this to have both the `initial` and `final` components in it as well, for the other part of this PR that got amputated.

Reply:
I made a custom class for this in: https://github.com/Qiskit/qiskit-terra/pull/8802/commits/d3bb3e04e1ba7533fbd1e1106893033979a09830 we can start to advertise it as a public thing along with a public layout attribute in 0.23.0